### PR TITLE
Autoparking

### PR DESCRIPTION
**Pull request by pjfernandecabo**
New parking spot with Qlearn algorithm. 
The parking spot is configured with white lines, walls, and static cars. The agent is a F1 car, equipped with cameras and lasers. The built-in RL algorithm is Qlearn.

### CODE REVIEW

File: rl_studio/agents/autoparking/train_qlearn.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+        ):
+            done = False
+            cumulated_reward = 0
+            step = 0
+            start_time_epoch = datetime.now()
+
+            observation = self.env.reset()
+            state = "".join(map(str, observation))
+
+            # ------- WHILE
```
Comment (rubenlucas93) [inline rl_studio/agents/autoparking/train_qlearn.py:216]: I would quit this comment. I think your code is clean enough to understand without this one

File: rl_studio/agents/autoparking/train_qlearn.py [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+                        episode=episode,
+                        episode_reward=cumulated_reward,
+                        steps=step,
+                        estimated_steps=self.estimated_steps,
+                        start_time=start_time,
+                        step_time=datetime.now()
+                        - timedelta(seconds=self.training_time),
+                    )
+                    if self.min_reward < cumulated_reward:
+                        self.min_reward = cumulated_reward
```
Comment (rubenlucas93) [inline rl_studio/agents/autoparking/train_qlearn.py:274]: Should it be named self.max_reward?

File: rl_studio/config/config.yaml [hunk 3]
```diff
@@ -85,6 +85,37 @@ agent:
         penal: 0
         min_reward: 1_000
         highest_reward: 100
+  autoparking:
```
Comment (rubenlucas93) [inline rl_studio/config/config.yaml:88]: This is something we may modify like we commented via slack.
We should make the config as generic as possible so we can indicate the parameters we want to run to our agent without need to indicate what agent we run.
I mean, instead of
f1:
camera ...
reward ...
autoparking:
camera ...
reward ....

We could have:
agent:
camera ...
reward ...

And just replace according to our preferences.

Let me what you think. It is just an idea.

**Review by rubenlucas93 — APPROVED** [review]

File: rl_studio/agents/autoparking/train_qlearn.py (hunk 1 above)
Comment (pjfernandecabo) [reply rl_studio/agents/autoparking/train_qlearn.py:216]: ok, it is true, when coding for research, we don't realize in personal comments

File: rl_studio/agents/autoparking/train_qlearn.py (hunk 2 above)
Comment (pjfernandecabo) [reply rl_studio/agents/autoparking/train_qlearn.py:274]: min_reward is correct. When the training has finished because it has reached the limit of defined steps, we check its result and only those that exceed the threshold defined in min_reward will be recorded.

File: rl_studio/config/config.yaml (hunk 3 above)
Comment (pjfernandecabo) [reply rl_studio/config/config.yaml:88]: yes, we have to think about how to better config.yaml
